src/gz/watchlist.c

```diff
@@ -262,13 +269,54 @@ static int destroy_proc(struct menu_item *item)
   return 0;
 }
 
+static int toggle_visibility_proc(struct menu_item *item,
+                      enum menu_callback_reason reason,
+                      void *data)
+{
+  struct item_data *item_data = data;
+  if (reason == MENU_CALLBACK_THINK) {
+    _Bool enabled;
+    if (item_data->members.size > 0) {
+      struct member_data *member_data = get_member(item_data, 0);
+      struct menu_item *watch = menu_userwatch_watch(member_data->userwatch);
+      enabled = menu_item_get_enabled(watch);
+    } else {
+      enabled = menu_checkbox_get(item);
+    }
+    menu_checkbox_set(item, enabled);
+  }
+  else if (reason == MENU_CALLBACK_SWITCH_ON) {
+    for (int i = 0; i < item_data->members.size; i++) {
+      struct member_data *member_data = get_member(item_data, i);
+      struct menu_item *watch = menu_userwatch_watch(member_data->userwatch);
+      menu_item_enable(watch);
+    }
+    menu_checkbox_set(item, 1);
+  }
+  else if (reason == MENU_CALLBACK_SWITCH_OFF) {
```

Comment:
The duplication between these last two conditional branches is a little bit jarring, but I couldn't think of a nicer way to do it. Advice appreciated! 😄